### ISSUE TITLE
no-undef rule fails when preceded by class definition containing member of type that wasn't defined

### ISSUE THREAD

**Issue by ChooKing** [oxc-project/oxc#11715]
When specifying this rule in .oxlintrc.json
```json
"rules": {    
    "no-undef: "error"    
  }
```
And using this minimal code example:
```ts
export class Foo{
    bar: notDefined;
}
const t = r + 1;
```
This causes a failure to identify that r is an undefined variable. Note that the notDefined type was never defined. 

Observations:
This bug only appears to arise when using a type annotation on a class member in which the annotated type was not defined. Moving this code outside of the class definition does not trigger the bug.

Running oxlint from the terminal catches this error, but the oxc plugin does not.

**Comment by ChooKing** [comment]
I just saw that no-undef is in nursery, but it works properly with the sample code when run from the terminal. It only fails via the oxc plugin.

**Comment by nrayburn-tech** [comment]
I'm seeing the same behavior with the CLI and IDE for Oxlint 1.1.0.

When using an undefined type, `r` does not produce an error. When using a defined type, `r` does produce an error that it is undefined.

@Boshen can you transfer this to the oxc repo? I don't seem to have access to do so.

**Comment by ChooKing** [comment]
It is very strange that now oxlint run from the terminal also has this problem. It definitely did not have it when I tested yesterday. I tried both 1.0.0 and 1.1.0 and they both have the problem now.  In previous tests, I used oxlint@latest instead of specifying a version number. Doing that now also shows the problem and didn't yesterday.

I have a second file in the directory which was also being linted, and that may have had an effect. Unfortunately, I lost the code that was in that file, so I cannot make a reproduction.
